cassovary-core/src/main/scala/com/twitter/cassovary/graph/UndirectedGraph.scala

```diff
@@ -0,0 +1,29 @@
+package com.twitter.cassovary.graph
+
+import com.twitter.cassovary.graph.StoredGraphDir.StoredGraphDir
+
+/*
+ * A graph where the outbound neighbors of each node equal the inbound neighbors.
+ * Given a DirectedGraph which stores the neighbors of each node (as outbound edges),
+ * this class wraps that graph in an undirected view.
+ */
+
+class UndirectedGraph(outboundGraph: DirectedGraph) extends DirectedGraph {
```

Comment:
An UndirectedGraph is a collection of UndirectedNodes so an access operation on it should return UndirectedNode. e.g., accessing an element of List[Dog] should return an element of type Dog, not an Animal. This is independent of what Dog might have that an Animal does not have because codes are always fluid.